feature/experimental/p4rt/otg_tests/p4rt_daemon_failure_test/p4rt_daemon_failure_test.go

```diff
@@ -0,0 +1,313 @@
+
+// configInterfaceDUT returns the OC Interface config for a given port.
+func configInterfaceDUT(i *oc.Interface, a *attrs.Attributes, dut *ondatra.DUTDevice) *oc.Interface {
+	i.Description = ygot.String(a.Desc)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if deviations.InterfaceEnabled(dut) {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if deviations.InterfaceEnabled(dut) && !deviations.IPv4MissingEnabled(dut) {
+		s4.Enabled = ygot.Bool(true)
+	}
+	s4a := s4.GetOrCreateAddress(a.IPv4)
+	s4a.PrefixLength = ygot.Uint8(ipv4PrefixLen)
+
+	return i
+}
+
+// configP4RTNode returns the OC P4RT Node Component for a given port.
+func configP4RTNode(nodeID string, deviceID uint64) *oc.Component {
+	c := &oc.Component{}
+	c.Name = ygot.String(nodeID)
+	c.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	c.IntegratedCircuit.NodeId = ygot.Uint64(deviceID)
+	return c
+}
+
+// configureDUT uses gNMI to configure port1 and port2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	d := gnmi.OC()
+	nodes := p4rtutils.P4RTNodesByPort(t, dut)
+
+	p1 := dut.Port(t, "port1")
+	i1 := &oc.Interface{Name: ygot.String(p1.Name())}
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(i1, &dutPort1, dut))
+
+	n1, ok := nodes[p1.ID()]
+	if !ok {
+		t.Fatal("P4RT node name for port1 not found.")
+	}
+	gnmi.Replace(t, dut, gnmi.OC().Component(n1).Config(), configP4RTNode(n1, deviceID1))
+
+	p2 := dut.Port(t, "port2")
+	i2 := &oc.Interface{Name: ygot.String(p2.Name())}
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(i2, &dutPort2, dut))
+	n2, ok := nodes[p2.ID()]
+	if !ok {
+		t.Fatal("P4RT node name for port2 not found.")
+	}
+	gnmi.Replace(t, dut, gnmi.OC().Component(n2).Config(), configP4RTNode(n2, deviceID2))
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, p1)
+		fptest.SetPortSpeed(t, p2)
+	}
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		fptest.AssignToNetworkInstance(t, dut, p1.Name(), deviations.DefaultNetworkInstance(dut), 0)
+		fptest.AssignToNetworkInstance(t, dut, p2.Name(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+}
+
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
+	otg := ate.OTG()
+	top := otg.NewConfig(t)
+
+	p1 := ate.Port(t, "port1")
+	atePort1.AddToOTG(top, p1, &dutPort1)
+
+	p2 := ate.Port(t, "port2")
+	atePort2.AddToOTG(top, p2, &dutPort2)
+
+	return top
+}
+
+// startTraffic generates traffic flow from source network to
+// destination network via atePort1 to atePort2.
+// Returns the flow object that it creates.
+func startTraffic(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config) gosnappi.Flow {
+	t.Helper()
+
+	otg := ate.OTG()
+	flow := top.Flows().Add().SetName("Flow")
+	flow.Metrics().SetEnable(true)
+	flow.TxRx().Device().SetTxNames([]string{atePort1.Name + ".IPv4"}).SetRxNames([]string{atePort1.Name + ".IPv4"})
```

Comment:
@ANISH-GOTTAPU : Rx port here should be atePort2?

Reply:
sorry, it was a typo. Updated now